### PR TITLE
Descriptive error message on create

### PR DESCRIPTION
**Pull request by blagoev**
This is related to https://github.com/realm/realm-object-store/pull/468

### CODE REVIEW

File: src/js_object_accessor.hpp [hunk 1]
```diff
@@ -249,7 +249,13 @@ struct Unbox<JSEngine, RowExpr> {
 template<typename T>
 template<typename U>
 U NativeAccessor<T>::unbox(ValueType value, bool create, bool update) {
-    return _impl::Unbox<T, U>::call(this, std::move(value), create, update);
+    try {
+        return _impl::Unbox<T, U>::call(this, std::move(value), create, update);
+    }
+    catch (TypeErrorException &ex) {
+        auto propertyType = ex.type();
+        throw InvalidPropertyException(propertyType, "unkonwn_property_name");
```
Comment (bdash) [inline src/js_object_accessor.hpp]: Typo: `"unkonwn_property_name"`

**Comment by bmunkholm** [comment]
@blagoev please also mention which issue this fixes. I believe this fixes #787?

**Comment by blagoev** [comment]
yes its #787. thanks for the correction

**Comment by blagoev** [comment]
closing this in favor of #1069

**Pull request closed without merging**